e2e/internal/keyexec/keyexec.go

```diff
@@ -0,0 +1,347 @@
+}
+
+func RemovePublicKeyring(t *testing.T) {
+	err := os.Remove(sypgp.PublicPath())
+	if err != nil {
+		t.Fatalf("Unable to remove public keyring: %v", err)
+	}
+}
+
+// RemoveSecretKeyring will delete your secret keyring.
+func RemoveSecretKeyring(t *testing.T) {
+	err := os.Remove(sypgp.SecretPath())
+	if err != nil {
+		t.Fatalf("Unable to remove secret keyring: %v", err)
+	}
+}
+
+func RemoveKeyring(t *testing.T) {
+	err := os.RemoveAll(sypgp.DirPath())
+	if err != nil {
+		t.Fatalf("Unable to remove keyring directory: %v", err)
+	}
+}
+
+// ImportKey will import a key from kpath.
+func ImportKey(t *testing.T, kpath string) (string, []byte, error) {
+	e2e.LoadEnv(t, &testenv)
+
+	argv := []string{"key", "import", kpath}
+	execKey := exec.Command(testenv.CmdPath, argv...)
+
+	cm := fmt.Sprintf("%s %s", testenv.CmdPath, strings.Join(argv, " "))
+
+	b, err := execKey.CombinedOutput()
+
+	return cm, b, err
+}
+
+// ImportPrivateKey will take a private key file (kpath) and import it.
+func ImportPrivateKey(t *testing.T, kpath string) (string, []byte, error) {
+	e2e.LoadEnv(t, &testenv)
+
+	c, err := expect.NewConsole()
+	if err != nil {
+		t.Fatal("Unable to start new console: ", err)
+	}
+	defer c.Close()
+
+	exportCmd := []string{"key", "import", kpath}
+
+	cmd := exec.Command(testenv.CmdPath, exportCmd...)
+	cmd.Stdin = c.Tty()
+
+	buf := bytes.NewBuffer(nil)
+	cmd.Stderr = buf
+	cmd.Stdout = buf
+
+	go func() {
+		c.ExpectEOF()
+	}()
+
+	err = cmd.Start()
+	if err != nil {
+		t.Fatal("Unable to start command: ", err)
+	}
+
+	// Send the passcode to singularity. The first one is the old
+	// password, the next two are the new passowrd.
+	c.Send("e2etests\n")
+	c.Send("e2etests\n")
+	c.Send("e2etests\n")
+
+	err = cmd.Wait()
+	cm := fmt.Sprintf("%s %s", testenv.CmdPath, strings.Join(exportCmd, " "))
+
+	return cm, buf.Bytes(), err
+}
+
+// ExportPrivateKey will import a private key from kpath.
+func ExportPrivateKey(t *testing.T, kpath, num string, armor bool) (string, []byte, error) {
+	e2e.LoadEnv(t, &testenv)
+
+	c, err := expect.NewConsole()
+	if err != nil {
+		t.Fatal("Unable to start new console: ", err)
+	}
+	defer c.Close()
+
+	exportCmd := []string{"key", "export", "--secret"}
+
+	if armor {
+		exportCmd = append(exportCmd, "--armor")
+	}
+
+	exportCmd = append(exportCmd, kpath)
+
+	outErr := bytes.NewBuffer(nil)
+
+	cmd := exec.Command(testenv.CmdPath, exportCmd...)
+	cmd.Stdin = c.Tty()
+
+	cmd.Stderr = outErr
+	cmd.Stdout = outErr
+
+	go func() {
+		c.ExpectEOF()
+	}()
+
+	err = cmd.Start()
+	if err != nil {
+		t.Fatalf("unable to run command: %v", err)
+	}
+
+	c.Send(num)
+	c.Send("e2etests\n")
+
+	err = cmd.Wait()
+	cm := fmt.Sprintf("%s %s", testenv.CmdPath, strings.Join(exportCmd, " "))
+
+	return cm, outErr.Bytes(), err
+}
+
+// RunKeyCmd will run a 'singularty key' command, with any args that are set in commands.
+func RunKeyCmd(t *testing.T, cmdPath string, commands []string, file, stdin string) (string, []byte, error) {
```

Comment:
why is the "file" argument special? It's used only once. The user of this function could pass it as part of the arguments (commands)

Reply:
Its slightly easier to read in the table tests, imo. But it could be part of the `commands []string`.

Reply:
Fixed now! :+1: